### PR TITLE
confirmation messages

### PR DESCRIPTION
**Pull request by IssaHanou**
### Issue
Closes GH-30, GH-72, GH-21, GH-78

## Contents
Implements confirmation messages between all components. 
- back-end can handle incoming confirmation messages
- front-end can handle incoming messages
- general message processing on front-end
- client libraries send their messages
- more general client library structure
- front-end can show 

## TODO
-  [x] client computers should send confirmation messages
- [x] back-end should pass along confirmation messages to front-end when necessary. 
- [ ] styling will be fixed in GH-71

This is a further implementation issue, whether instructions from front-end should be passed to client computers with instructor front-end or back-end. this can be implemented later.

### CODE REVIEW

File: front-end/src/app/components/device/device.component.ts [hunk 1]
```diff
@@ -1,68 +1,26 @@
...
-            message.topic
-        );
-        this.onMessageArrived(message.payload);
-      });
-    console.log("subscribed to topic: " + this.topicname);
-  }
-
-  constructor(private mqttService: MqttService) {
-    this.subscribeNewTopic();
+  ngOnInit() {
```
Comment (mrlomar) [inline front-end/src/app/components/device/device.component.ts]: can be removed?

File: back-end/src/sciler/handler/handler.go [hunk 2]
```diff
@@ -144,7 +192,7 @@ func (handler *Handler) onInstructionMsg(raw Message) {
 	logrus.Info("instruction message received from: ", raw.DeviceID)
 	if raw.Contents["instruction"] == "test all" && raw.DeviceID == "front-end" { // TODO maybe switch again
 		message := Message{
-			DeviceID: "back-end",
```
Comment (mrlomar) [inline back-end/src/sciler/handler/handler.go]: This should be back-end, this makes the logs unreadable.
TODO: make additional field: confirmation-sender, to which a confirmation is sent if not empty

File: cc_library/src/scripts/controlboard.py [hunk 3]
```diff
@@ -109,13 +114,13 @@ def blink(self, data):
         GPIO.output(led, GPIO.LOW)
         time.sleep(interval)
 
-    def turn_off(self, data):
-        led = getattr(self, data.get("led"))
-        GPIO.output(led, GPIO.LOW)
+    def turn_off(self):
```
Comment (GwennanS) [inline cc_library/src/scripts/controlboard.py]: We want to be able to control all leds separately

File: back-end/src/sciler/handler/handler.go (hunk 2 above)
Comment (GwennanS) [reply back-end/src/sciler/handler/handler.go]: made issue GH-80

**Pull request merged**